migrations/scripts/generate_schema

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Generate the converter's base intermediate database migration file from
+# the core database state and YAML configuration in schema.yml
+# Invoke from core root directory as `./migrations/scripts/generate_schema`
+# It accepts an optional command line argument for the output file path which
+# overrides the path configured in schema.yml
+
+puts "Loading application..."
+require_relative "../../config/environment"
+
+module Migrations
+  class GenerateSchema
+    def initialize(opts = {})
+      config = YAML.load_file(File.join(__dir__, "schema.yml"), symbolize_names: true)
+
+      @core_db_connection = ActiveRecord::Base.connection
+      @output_stream = StringIO.new
+      @indirectly_ignored_columns = Hash.new { |h, k| h[k] = [] }
+
+      @output_directory = config[:output_directory]
+      @output_filename = config[:output_filename]
+      @output_file_path = opts[:output_file_path]
+      @table_configs = config[:tables]
+      @column_configs = config[:columns]
+
+      @configured_table_names = @table_configs&.keys&.sort || []
+      @global_column_ignore_list = @column_configs[:ignore] || []
+    end
+
+    def run
+      puts "Generating base converter migration file for Discourse #{Discourse::VERSION::STRING}"
+
+      generate_header
+      generate_tables
+      generate_indirectly_ignored_columns_log
+      generate_migration_file
+
+      puts "", "Done"
+    end
+
+    private
+
+    def generate_header
+      @output_stream.puts <<~HEADER
+        /*
+         This file is auto-generated from the Discourse core database schema. Instead of editing it directly,
+         please update the `schema.yml` configuration file and re-run the `generate_schema` script to update it.
+        */
+      HEADER
+    end
+
+    def generate_tables
+      puts "Generating tables..."
+
+      @configured_table_names.each do |name|
+        raise "Core table named '#{name}' not found" unless valid_table?(name)
+
+        generate_table(name)
+      end
+    end
+
+    def generate_indirectly_ignored_columns_log
```

Comment:
Should this method write a comment to the output file that explains what those columns at the end of the file are?

Reply:
Yes, I'll add that.